### DISCUSSION TITLE
Fields Arrays with validate if min fields

### DISCUSSION THREAD

**Discussion by majweb** [react-hook-form/react-hook-form#2605] [Q&A]
I warmly welcome.
I am using FieldsArray.
This is an example from the documentation.
https://codesandbox.io/s/react-hook-form-fieldsarray-yup-validation-min-length-57qtb?file=/src/index.js:849-852

I have quit yup and want to use native validation.
The problem is validation and no error if no project is created.

My code
https://codesandbox.io/s/vigorous-almeida-ngp5r?fontsize=14&hidenavigation=1&theme=dark

I see the problem in useeffect but I can be wrong ..

**Comment by bluebill1049** [comment]
for minLength take a look at this example: 

https://github.com/react-hook-form/react-hook-form/tree/master/examples

`Field Array Min Length`

**Comment by majweb** [comment]
I added your link in my question

**Comment by bluebill1049** [comment]
is it working for u?

**Comment by majweb** [comment]
No

**Comment by bluebill1049** [comment]
what bout this example: https://codesandbox.io/s/react-hook-form-fieldsarray-yup-validation-min-length-forked-kb90b?file=/src/index.js?

**Comment by majweb** [comment]
how pass invalid className?

```
      <input
            ref={register()}
            name={`questions[${questionIndex}].text`}
            control={control}
            defaultValue=""
            className={`form-control ${`errors.questions[${questionIndex}].text` ? "is-invalid" : ""}`}
          />
```

**Comment by bluebill1049** [comment]
that's looks right for individual input error.

**Reply by majweb** [reply]
Looks right but still in-valid class:
https://pastebin.com/AR8SuTmA

**Reply by bluebill1049** [reply]
what do you mean in-valid class? is the errors object correct?

**Comment by webhasan** [comment]
I like to throw an error when there is no field. Must have to have at least one item field
My yup schema validation: 
``` js
const schema = object({
   items: array(
      object({
         description: string().required('Project description is required').min(3, 'Invoice number must be at least 3 characters'),
         price: number().required('Price is required').positive('Price must be positive number.').integer(),
      })
   ) .min(1, 'Must need at least one items')
}).required();
```



How I checked the error: 
```js
    {fields.map((field, index) => {
        return (
           <Grid container key={field.id} alignItems="center" spacing={1} data-test={`invoice-item-${index}`}>
              <Grid item xs={1}>
              <IconButton aria-label="delete" onClick={() => remove(index)}>
                 <DeleteIcon color="error"/>
              </IconButton>
              </Grid>
              <Grid item xs={8}>
                 <TextField
                    margin="normal"
                    fullWidth
                    label="Description*"
id={`description-${field.id}`}
                    inputProps={{ "data-test": "invoice-item-description" }}
                    error={!!errors.items?.[index]?.description}
                    helperText={
                       errors.items?.[index]?.description && (
                          <span data-test="invoice-item-description-error">
                             {errors.items?.[index]?.description?.message}
                          </span>
                       )
                    }
                    {...register(`items.${index}.description`)}
                 />
              </Grid>
              <Grid item xs={3}>
                 <TextField
                    margin="normal"
                    fullWidth
                    label="Pice*"
                    id={`price-${field.id}`}
                    type="number"
                    inputProps={{ "data-test": "invoice-item-value" }}
                    error={!!errors.items?.[index]?.price}
                    helperText={
                       errors.items?.[index]?.price && (
                          <span data-test="invoice-item-value-error">
                             {errors.items?.[index]?.price?.message}
                          </span>
                       )
                    }
                    {...register(`items.${index}.price`)}
                 />
              </Grid>
           </Grid>
        )
     })}
					
<Button onClick={() => {append({description: '', price: 0}); clearErrors('items')}}>Add Item + </Button>
{errors.items  && <FormHelperText error>{errors.items.message as string}</FormHelperText>}
```

The very last line code perfectly works as expected but unfortunately, I am getting typescript errors as for: `errors.items.message`
```txt
Property 'message' does not exist on type '{ description?: FieldError | undefined; price?: FieldError | undefined; }[]'.ts(2339)
```

[Check screenshot: ](https://prnt.sc/k5TDxRG4qnFZ)

**Comment by udayanss** [comment]
Hello @erikras ,I have two buttons inside FieldsArray, how do i send two different parameters while adding each fields
<button
          className="text-primaryLight cursor-pointer button mx-2"
          // onClick={() => fields.push({section:'product'})}
          onClick={() => fields.push({},{section:'product'})}
       
        >
          Add Product Section
        </button>
        <button
          className="text-primaryLight cursor-pointer button mx-2"
          // onClick={() => fields.push({section:'banner'})}
          onClick={() => fields.push({},{section:'banner'})}
     
        >
          Add Banner Section
        </button>
